crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -296,24 +295,29 @@ impl Target {
         match &instruction {
             TargetOperation::Variadic(_) => {
                 qargs_val = PropsMap::with_capacity(1);
-                qargs_val.extend([(None, None)]);
+                qargs_val.extend([(smallvec![], None)]);
             }
             TargetOperation::Normal(normal) => {
                 if let Some(mut properties) = properties {
                     qargs_val = PropsMap::with_capacity(properties.len());
                     let inst_num_qubits = normal.operation.view().num_qubits();
-                    if properties.contains_key(None) {
+                    if properties.contains_key(&None) {
                         self.global_operations
                             .entry(inst_num_qubits)
                             .and_modify(|e| {
                                 e.insert(name.to_string());
                             })
                             .or_insert(HashSet::from_iter([name.to_string()]));
                     }
-                    let property_keys: Vec<Option<Qargs>> =
-                        properties.keys().map(|qargs| qargs.cloned()).collect();
+                    let property_keys: Vec<Qargs> = properties
+                        .keys()
+                        .map(|qargs| match qargs {
+                            Some(qargs) => qargs.clone(),
+                            None => smallvec![],
+                        })
+                        .collect();
```

Comment:
I realize this was pre-existing, but why do we need to collect here? Like I see why it's needed in the current because you're iterating over properties' keys and mutating properties. But I feel like this could be refactored to not need to mutate anything in properties. Maybe for a followup though.

Reply:
Solved in f3e88b1